ciao-controller/server.go

```diff
@@ -69,6 +70,13 @@ func (h *clientCertAuthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 	}
 
 	r = r.WithContext(service.SetTenantID(r.Context(), tenantFromVars))
+	if tenantFromVars != "" {
+		err := h.Controller.confirmTenant(tenantFromVars)
+		if err != nil {
+			http.Error(w, "Error confirming tenant", http.StatusInternalServerError)
```

Comment:
Will this end up returning a different error code?  Should we return errorResponse(err) instead.  Note it's not entirely clear to me that errorResponse works properly with wrapped errors.

Reply:
I guess there's an inherent race here.  Just because the tenant is confirmed at this point of the transaction doesn't mean that the tenant will still exist when we actually create the volume.   Someone else could be calling DeleteTenant at the same time.  As far as I can tell, this race already exists in the code, so this is a question rather than a comment.  Will moving the tenant check here make it harder to fix the race, should it need fixing?  Maybe it makes it easier.